### PR TITLE
feat: official clang-cl support

### PR DESCRIPTION
**Pull request by Jaskowicz1**
This PR fixes clang-cl, allowing it to now compile and work.

I am in a slight doubt about this fix. Whilst it works, i'm not too sure if it's correct. It looks like the each avx.h goes into a isa_detection.h file, with the `#if` check. Clang-cl only knows of `__m128` so I don't know if we need the `#if` check in `avx.h` but I'll let others decide that.

## Code change checklist

- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR.
- [ ] I have ensured that I did not break any existing API calls.
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

**Comment by Jaskowicz1** [comment]
Hm, this seems to be the appropriate fix as far as I can see. It seems that `immintrin.h` is broken via clang-cl with these lines:
```cpp
#if !(defined(_MSC_VER) || defined(__SCE__)) || __has_feature(modules) ||      \
    defined(__SSE__)
#include <xmmintrin.h>
#endif

#if !(defined(_MSC_VER) || defined(__SCE__)) || __has_feature(modules) ||      \
    defined(__SSE2__)
#include <emmintrin.h>
#endif
```
That results in this:
image

**Comment by Jaskowicz1** [comment]
Slightly confused now. clang-cl can successfully use the template bot with these changes, but building from source fails completely.

**Comment by Jaskowicz1** [comment]
With coro, It looks as if `set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /await:strict")`  on line 298 in `library/CMakeLists.txt` doesn't actually apply `/await:strict` in clang-cl. However, it does read it.

File: include/dpp/isa/avx2.h [hunk 1]
```diff
@@ -20,7 +20,11 @@
  ************************************************************************************/
 #pragma once
 
-#if defined _MSC_VER || defined __GNUC__ || defined __clang__
+/* 
+ * If the user is using clang-cl, both _MSC_VER and __clang__ are defined. However, clang-cl breaks using this as it knows nothing above __m128.
+ * So, we make sure that both aren't defined, and only do this if _MSC_VER, __GNUC__, or __clang__, is defined.
+ */
+#if (!defined _MSC_VER && !defined __clang__) && (defined _MSC_VER || defined __GNUC__ || defined __clang__)
```
[the remark this answers is not in the document]
Comment (braindigitalis) [reply include/dpp/isa/avx2.h:27]: so, never on clang?

IF
msc or clang ARENT defined

AND

msc, gnuc or clang ARE defined

these can never both be true for CLANG?

**Review by Mishura4 — APPROVED** [review]
Hm yes, I have reviewed these changes and I must quite say, I find myself satisfied with these changes yes, good stuff Archie! :catsuit:

**Comment by Jaskowicz1** [comment]
Now marked as ready for review.

**Comment by Mishura4** [comment]
Sike

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**